cdap-master/src/main/java/co/cask/cdap/data/tools/StreamStateStoreUpgrader.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data.tools;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.common.namespace.NamespacedLocationFactory;
+import co.cask.cdap.data.stream.StreamUtils;
+import co.cask.cdap.data2.util.TableId;
+import co.cask.cdap.data2.util.hbase.HBaseTableUtil;
+import co.cask.cdap.proto.Id;
+import com.google.inject.Inject;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.twill.filesystem.LocationFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+
+/**
+ * Upgrades stream state store table.
+ *
+ * Modifies all row keys from "namespace:foo/stream:bar".getBytes() to "stream:foo.bar".getBytes().
+ */
+public class StreamStateStoreUpgrader extends AbstractQueueUpgrader {
+  private static final Logger LOG = LoggerFactory.getLogger(StreamStateStoreUpgrader.class);
+
+  @Inject
+  public StreamStateStoreUpgrader(LocationFactory locationFactory, HBaseTableUtil tableUtil,
+                                  NamespacedLocationFactory namespacedLocationFactory, Configuration conf) {
+    super(locationFactory, namespacedLocationFactory, tableUtil, conf);
+  }
+
+  @Override
+  protected TableId getTableId() {
+    return StreamUtils.getStateStoreTableId(Id.Namespace.DEFAULT);
```

Comment:
I think AbstractQueueUpgrader should now be refactored to upgrade all stream state store tables - not just the one in Default namespace.
Previously - for the upgrade tool in CDAP 2.8 - only the table in default namespace existed before upgrade, and so it only had to operate on that one table.